mParticle-Apple-SDK/Utils/NSDictionary+MPCaseInsensitive.m

```diff
@@ -72,8 +72,7 @@ - (id)valueForCaseInsensitiveKey:(NSString *)key {
             transformedDictionary[key] = [[NSString alloc] initWithData:obj encoding:NSUTF8StringEncoding];
         } else {
             MPILogError(@"Data type is not supported as an attribute value: %@ - %@", obj, [[obj class] description]);
-            NSAssert([obj isKindOfClass:[NSString class]], @"Data type is not supported as an attribute value");
-            return;
+            transformedDictionary[key] = [obj description];
```

Comment:
I remember we decided on this (using `description`) since that's what Android is doing, though this will result in a different string than Android. I think it would be ideal to use a JSON string on both platforms, but I think we decided that was out of scope for this work and could be something we do in the future...

Reply:
I actually did this initially but was checking count (example below) which I didn't realize was what was making me not hit that statement. Thanks for making me take another look.

```
 else if ([obj isKindOfClass:[NSDictionary class]] && [(NSDictionary *)obj count] > 0) {
            transformedDictionary[key] = [obj description];
        } else if ([obj isKindOfClass:[NSMutableDictionary class]] && [(NSDictionary *)obj count] > 0) {
            transformedDictionary[key] = [obj description];
        } else {
            MPILogError(@"Data type is not supported as an attribute value: %@ - %@", obj, [[obj class] description]);
            NSAssert([obj isKindOfClass:[NSString class]], @"Data type is not supported as an attribute value");
            return;
        }
        ```